### ISSUE TITLE
Expand of system enviromentVariables in metadata src

### ISSUE THREAD

**Issue by bqstony** [dotnet/docfx#4983]
**Operation System**: (`Windows` or `Linux` or `MacOS`)
Windows 10

**DocFX Version Used**: 
docfx.console v2.44.0

**Template used**: (`default` or `statictoc` or contain custom template)
statictoc

**Steps to Reproduce**:
I  Pack the source code to our internal nuget packages. This source code i would like to consume for creation of api documentation for appliation and included packages.

When i define
```
"metadata": [
    {
      "src": [
        {
          "src": "../Source",
          "files": [
            "**/*.csproj"
          ]
        },
        {
          "src": "%userprofile%\\.nuget\\packages",
          "files": [
            "mbc.pcs.net/2.0.0/src/Mbc.Pcs.Net/Mbc.Pcs.Net.csproj"
          ]
        },
        {
          "src": "C:\\Users\\mihe\\.nuget\\packages",
          "files": [
            "mbc.pcs.net/2.0.0/src/Mbc.Pcs.Net/Mbc.Pcs.Net.csproj"
          ]
        }
      ],
      "dest": "api",      
      "disableGitFeatures": false,
      "disableDefaultFilter": false,
      "properties": {
        "TargetFramework": "net471"
      }
    }
  ]
```
The "src": "C:\\Users\\mihe\\.nuget\\packages", does work. but not "src": "%userprofile%\\.nuget\\packages",

**Expected Behavior**:

When i use Enviroment variable like `%userprofile%` it should expand it to C:\\Users\\mihe\\ and should result in `C:\\Users\\mihe\\.nuget\\packages`

**Actual Behavior**:

It combines the Project root + %userprofile% to `C:\\source\\ProjectX\\Docs\\%userprofile%\\.nuget\\packages\`

From log:
```
{"message":"No files are found with glob pattern mbc.pcs.net/2.0.0/src/Mbc.Pcs.Net/Mbc.Pcs.Net.csproj, excluding <none>, under directory \"C:\\source\\ProjectX\\Docs\\%userprofile%\\.nuget\\packages\"","source":"MetadataCommand.ExtractMetadata","date_time":"2019-08-13T07:07:20.4325616Z","message_severity":"info","correlation_id":"AD12ABAE-AEEF-4DB5-8F8F-490669C4AECF.1.1.2"}
```

**Comment by bqstony** [comment]
@superyyrrzz can you give me a starting Point, wehre the behavior should be implemented, then i try to create a PR.

**Comment by superyyrrzz** [comment]
This seems like a one-line code change. You need to `Environment.ExpandEnvironmentVariables` in `src` before evaluating the globs. See: https://github.com/dotnet/docfx/blob/734b6ea36773cbafda2cda2c2fe71c44959face7/src/docfx/Glob/GlobUtility.cs#L31

PR is welcomed 😄

**Comment by KalleOlaviNiemitalo** [comment]
The related documentation change might be more than one line though 😃

**Comment by KalleOlaviNiemitalo** [comment]
The Path.IsPathRooted call should not be necessary. Path.Combine will detect if expandedSourceFolder is rooted and not let baseDirectory affect the return value in that case. <https://github.com/microsoft/referencesource/blob/74eb1593e09a636270482f1c0525aabdccb1f364/mscorlib/system/io/path.cs#L1295-L1296>

**Comment by KalleOlaviNiemitalo** [comment]
Can this be closed now that #4994 was merged?

**Comment by superyyrrzz** [comment]
I will close this once the new version releases on 8/22. @KalleOlaviNiemitalo
